### PR TITLE
Improve tab handling in JS transformer

### PR DESCRIPTION
**Pull request by colinking**
## Description

Updates the JS transformer to handle task code that uses tabs for indentation. I ran into this while testing, as the code we generate from `airplane tasks init` uses tabs.

## Test plan

Unit tests pass.

### CODE REVIEW

**Comment by colinking** [comment]
Current dependencies on/for this PR:
* main
  * **PR #567** <a href="https://app.graphite.dev/github/pr/airplanedev/lib/567" target="_blank">Graphite</a> 
    * **PR #568** <a href="https://app.graphite.dev/github/pr/airplanedev/lib/568" target="_blank">Graphite</a> 
      * **PR #571** <a href="https://app.graphite.dev/github/pr/airplanedev/lib/571" target="_blank">Graphite</a> 
        * **PR #572** <a href="https://app.graphite.dev/github/pr/airplanedev/lib/572" target="_blank">Graphite</a> 
          * **PR #580** <a href="https://app.graphite.dev/github/pr/airplanedev/lib/580" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/airplanedev/lib/580?utm_source=stack-comment).

File: pkg/runtime/javascript/transformer/transformer.ts [hunk 1]
```diff
@@ -385,3 +386,38 @@ const assertNever = (value: never): never => {
...
+    // This is necessary to avoid infinite loops with zero-width matches
+    if (m.index === regex.lastIndex) {
+      regex.lastIndex++;
+    }
+
+    // The result can be accessed through the `m`-variable.
+    m.forEach((match) => {
+      if (match.includes(" ")) spaceLines++;
+      if (match.includes("\t")) tabLines++;
+    });
```
Comment (ameyakhare) [inline pkg/runtime/javascript/transformer/transformer.ts:413]: maybe some short-circuiting here if you encounter both spaces and tabs

File: pkg/runtime/javascript/transformer/transformer.ts [hunk 2]
```diff
@@ -385,3 +386,38 @@ const assertNever = (value: never): never => {
...
+ */
+const getIndentation = (contents: string): "tabs" | "spaces" | undefined => {
+  if (!contents) {
+    return undefined;
+  }
+  const regex = /^[ \t]*/gm;
+  let tabLines = 0;
+  let spaceLines = 0;
+  let m: RegExpExecArray | null;
+  while ((m = regex.exec(contents)) !== null) {
```
Comment (ameyakhare) [inline pkg/runtime/javascript/transformer/transformer.ts:403]: do we really have to go through the whole file? can we just assume whatever there is at the beginning is the choice for the whole file? or maybe the first few whitespaces?

**Review by ameyakhare — APPROVED** [review]

File: pkg/runtime/javascript/transformer/transformer.ts (hunk 2 above)
Comment (colinking) [reply pkg/runtime/javascript/transformer/transformer.ts:403]: The file content is already loaded into memory, so this check should be very fast. I don't see any reason to optimize it further.

**Pull request merged**